src/bootstrap/bin/rustdoc.rs

```diff
@@ -53,10 +53,30 @@ fn main() {
         arg.push(&linker);
         cmd.arg(arg);
     }
-    if let Ok(no_threads) = env::var("RUSTDOC_LLD_NO_THREADS") {
-        cmd.arg("-Clink-arg=-fuse-ld=lld");
-        cmd.arg(format!("-Clink-arg=-Wl,{no_threads}"));
+
+    if let Ok(lld) = env::var("RUSTDOC_HOST_LLD") {
```

Comment:
The logic below probably shouldn't live in the binary wrappers.
`RUSTC_HOST_LLD`/`RUSTDOC_HOST_LLD` (or rather its replacement) should contain an already pre-cooked command line string.
Then this logic can be shared between 2 or even 3 places in which it's used.

Reply:
I know that it's not pretty, but if I passed the command line through the env var, I'd have to encode/decode it somehow, I think. Because internally this uses `cmd.arg`, but I would need to pass it several arguments (e.g. `-Clink-flavor=gnu-lld-cc -Clink-self-contained=+linker`).